File: season_1/programming_lab/TDE_01/exercise_06.py
try:
    numero = int(input("Digite um número: "))

    if numero < 0:
        print("Números negativos não são quadrados perfeitos.")
    else:
        raiz = int(numero ** 0.5)

        if raiz * raiz == numero:
            print("É um quadrado perfeito.")
        else:
            print("Não é um quadrado perfeito.")

except ValueError:
    print("Erro: digite um número inteiro válido.")

File: season_1/programming_lab/TDE_01/exercise_06.c
#include <stdio.h>
#include <math.h>

int main() {
    int numero;

    printf("Digite um numero: ");

    // Verifica se a leitura foi válida
    if (scanf("%d", &numero) != 1) {
        printf("Erro: digite um numero inteiro valido.\n");
        return 1;
    }

    if (numero < 0) {
        printf("Numeros negativos nao sao quadrados perfeitos.\n");
    } else {
        int raiz = (int) sqrt(numero);

        if (raiz * raiz == numero) {
            printf("E um quadrado perfeito.\n");
        } else {
            printf("Nao e um quadrado perfeito.\n");
        }
    }

    return 0;
}